lib/httpServer/transports/http/index.js

```diff
@@ -615,6 +638,57 @@ function safeDecodePath(path) {
 	}
 }
 
+/**
+ * Encapsulate the request object
+ *
+ * This function currently only takes care of
+ * checking the HTTP POST size.
+ *
+ * @param {*} req HTTP request
+ */
+function setRequestFilters(req) {
+	var realOn = req.on.bind(req);
+	var byteSize = 0;
+
+	function isRequestTooLarge() {
+		return byteSize > maxRequestSize;
+	}
+
+	req.on('data', function (chunk) {
+		byteSize += chunk.length;
+
+		if (isRequestTooLarge()) {
+			req.emit('error', new RequestError('Entity too large', req));
+			req.removeAllListeners();
```

Comment:
`req.removeAllListeners('data');` just to be sure? This doesn't feel (but may well be) side-effect free.

Reply:
We remove all event listeners because we also want to remove 'end' right now; we don't want 'end' to trigger if we have aborted.

Going through the list of events, I'd say removing everything *should* be safe enough - maybe even the safest thing to do. Can you see any events we'd actually want to fire on the stream post-error emission?

Reply:
No, it was more about future proofing and the fact that "get rid of all-the-things" is a bit of a broad sweeping approach. I doubt any Node documentation would ever argue that's a good idea to do on Node's internal APIs. To me, it's a bit like using underscored properties on Node internal APIs. Just because you can, doesn't mean you should (in that Node's internals may depend on you not "misbehaving").

Reply:
Right, indeed Node's internal never use it (only case I found is https://github.com/nodejs/node/blob/c0d858f8bb8ba5212548da2fba6a7bc02db0462b/deps/npm/lib/search/all-package-metadata.js#L184 but its not  really relevant).

So I guess removing 'data' and 'end' would make more sense.